pkg/datapath/maps/map.go

```diff
@@ -175,6 +175,15 @@ func (ms *MapSweeper) RemoveDisabledMaps() {
 		}...)
 	}
 
+	if !option.Config.EnableBPFMasquerade && !option.Config.EnableIPv4Masquerade {
+		maps = append(maps, []string{"cilium_snat_v4_external"}...)
+	}
+
+	if (option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled) ||
+		(option.Config.KubeProxyReplacement == option.KubeProxyReplacementPartial && !option.Config.EnableNodePort) {
```

Comment:
@brb Would it be enough to check `!option.Config.EnableNodePort` here given it is disabled if `option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled`?

Reply:
Yep, that's a good suggestion. Also, BPF masquerade depends on it, so it's enough to check only for NodePort.

Reply:
Going to change that. Thanks guys.